app/src/services/webtoon/webtoonService.js

```diff
@@ -3,14 +3,30 @@
 const webtoonRepository = require("../../repositories/webtoon/webtoonRepository");
 
 class WebtoonService {
-  async getAllWebtoons() {
+  async getWebtoons(filters) {
     try {
-      const webtoon = await webtoonRepository.getAllWebtoons();
-      return {
-        status: 200,
-        success: true,
-        data: { webtoon },
-      };
+      const { day } = filters;
+      if (!day) {
+        const allWebtoons = await webtoonRepository.getAllWebtoons();
+        return {
+          status: 200,
+          success: true,
+          data: { allWebtoons },
+        };
+      } else if (day) {
+        const webtoonsByDay = await webtoonRepository.getWebtoonsByDay(day);
+        return {
+          status: 200,
+          success: true,
+          data: { webtoonsByDay },
+        };
+      } else {
+        return {
+          status: 400,
+          success: false,
+          data: { message: "지원하지 않는 필터 조건입니다." },
+        };
+      }
```

Comment:
변경요청: 하나의 API의 응답 포멧은 항상 동일해야합니다. 지금은 요청 파라미터에 따라 allWentoons, webtoonsByDay 두가지 케이스로 응답되는 구조같아요

그리 복잡하지 않은 로직인데 구현이 복잡해진거같습니다.

1. 첫번째 if문은 day가 존재하지 않는 경우, 두번째 if문은 day가 존재하는 경우, 3번째 if문은 그 무엇도 아닌 경우 -> 3번째에 해당하는 경우는 없으므로 제거해도 됨
2. [https://eslint.org/docs/latest/rules/no-else-return](no else return) 이라는 규칙이 있는데 이를 적용하면 인덴트(들여쓰기)를 줄일 수 있어서 가독성이 좋아집니다(물론 명시적으로 if 규칙을 표현하기 위해 규칙을 적용하지 않는 경우도 있어요)

아래는 위 규칙을 기반으로 리펙토링한 코드입니다

```suggestion
      const { day } = filters;

      if (!day) {
        const allWebtoons = await webtoonRepository.getAllWebtoons();
        return {
          status: 200,
          success: true,
          data: { allWebtoons },
        };
      }

      const webtoonsByDay = await webtoonRepository.getWebtoonsByDay(day);
      return {
        status: 200,
        success: true,
        data: { webtoonsByDay },
      };
```

좀 더 짧게 작성하고 싶으시다면 아래처럼도 가능할거같아요

```js
      const webtoons = day
        ? await webtoonRepository.getWebtoonsByDay(day)
        : await webtoonRepository.getAllWebtoons();

      return {
        status: 200,
        success: true,
        data: { webtoons },
      };
```

Reply:
위에서 설명드린 것처럼, 추후 sort 쿼리 파라미터로 view, like, updated와 같은 정렬 조건 들어올 것을 고려해 if - else if - else 구조로 작성했었습니다. 하지만 말씀해주신 대로 지금 코드에서는 구현이 다소 복잡해졌고, no-else-return 규칙도 위반하게 된 것 같습니다. 특히 번째 else문은 테스트 과정에서  잘못된 입력을 고려해 넣었는데, 다시 생각해보니 실제 상황에서는 발생할 수 없는 조건이었습니다. 앞으로는 현재 이슈에 집중해 더 간결하고 명확한 코드로 작성하겠습니다. 피드백 감사합니다!

Reply:
여러 상황에 따라 별도의 메서드를 호출해야한다면 if - else if 구조가 적절한거같습니다. 상항연산자는 2개 이상 넘어가면 가독성을 해치더라고요.
no-else-return은 꼭 따라야하는 규칙은 아니라 취향껏 사용하시면 좋을거같습니다. 컨벤션의 영역으로 가면 팀원간 논의 후 결정을 해야하겠구요.
모든 논리적인 케이스에 대해 처리하시려고 하는것도 좋지만 처리해야하는 케이스가 맞는지 한번 더 고민하는 단계가 있다면 좋은 코드에 도움이 될거같아요